dcicutils/ecr_utils.py

```diff
@@ -77,6 +98,166 @@ def extract_ecr_password_from_authorization(*, authorization):
         """
         return base64.b64decode(authorization['authorizationToken']).replace(b'AWS:', b'').decode('utf-8')
 
+    def get_images_descriptions(self,
+                                digests: Optional[List[str]] = None,
+                                tags: Optional[List[str]] = None,
+                                limit: Optional[Union[int, str]] = IMAGE_LIST_DEFAULT_COUNT_LIMIT):
+        """
+        Args:
+            digests: a list of image digests (each represented as a string in sha256 format)
+            tags: a list of strings (each being a potential or actual image tag).
+            limit: an int indicating a limit on how many results to return,
+                   a string indicating a limiting tag,
+                   or None indicating no limit.
+
+        Returns:
+            a dictionary with keys 'descriptions' (the primary result), 'count' (the number of descriptions),
+            and 'total' (the total number of registered images in the catalog).
+            The descriptions are a list of individual dictionaries with keys that include, at least,
+            'imagePushedAt', 'imageTags', and 'imageDigest'.
+        """
+        next_token = None
+        image_descriptions = []
+        # We don't know what order they are in, so we need to pull them all down,
+        # and only hen sort them before applying the limit.
```

Comment:
typo: then - also very strange you can't pull them down in order, perhaps we should just tag all of them both with version and "latest" so they come down in a reasonable order?